dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/loadbalance/RandomLoadBalance.java

```diff
@@ -34,13 +34,13 @@
     @Override
     protected <T> Invoker<T> doSelect(List<Invoker<T>> invokers, URL url, Invocation invocation) {
         int length = invokers.size(); // Number of invokers
-        int totalWeight = 0; // The sum of weights
         boolean sameWeight = true; // Every invoker has the same weight?
-        for (int i = 0; i < length; i++) {
+        int firstWeight = getWeight(invokers.get(0), invocation);
```

Comment:
out of bound without checking

Reply:
I think it is not necessary to check because the select method of AbstractLoadBalance that invoke the method has check the null pointer and the size of the invoker list. If the size of the list is 0, it will not reach here.